src/js/tracks/text-track.js

```diff
@@ -102,39 +99,22 @@ const loadTrack = function(src, track) {
  * };
  *
  * @param {Object=} options Object of option names and values
- * @extends EventTarget
+ * @extends Track
  * @class TextTrack
  */
-class TextTrack extends EventTarget {
+class TextTrack extends Track {
   constructor(options = {}) {
-    super();
-    if (!options.tech) {
-      throw new Error('A tech was not provided.');
-    }
-
-    let tt = this;
-
-    if (browser.IS_IE8) {
-      tt = document.createElement('custom');
-
-      for (let prop in TextTrack.prototype) {
-        if (prop !== 'constructor') {
-          tt[prop] = TextTrack.prototype[prop];
-        }
-      }
-    }
-
-    tt.tech_ = options.tech;
-
+    options.kind = TextTrackEnum.TextTrackKind[options.kind] || 'subtitles';
```

Comment:
we shouldn't be mutating the options object. We should be using new local variables.